src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForLoadTests.cs

```diff
@@ -0,0 +1,741 @@
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+            Dim xml = """"
+            Dim doc = New XmlDocument()
+            doc.XmlResolver = Nothing
+            doc.Load(xml)
+        End Try
+    End Sub
+End Class",
+                GetCA3075LoadBasicResultAt(13, 13, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXmlDocumentLoadInAsyncAwaitShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+ using System.Threading.Tasks;
+ using System.Xml;
+
+    class TestClass
+    {
+        private async Task TestMethod()
+        {
+            await Task.Run(() => {
+                var xml = """";
+                var doc = new XmlDocument();
+                doc.XmlResolver = null;
+                doc.Load(xml);
+            });
+        }
+
+        private async void TestMethod2()
+        {
+            await TestMethod();
+        }
+    }",
+                GetCA3075LoadCSharpResultAt(13, 17, "Run")
+            );
+
+            VerifyBasic(@"
+Imports System.Threading.Tasks
+Imports System.Xml
+
+Class TestClass
+    Private Function TestMethod() As Task
+        Await Task.Run(Function() 
+        Dim xml = """"
+        Dim doc = New XmlDocument()
+        doc.XmlResolver = Nothing
+        doc.Load(xml)
+
+End Function)
+    End Function
+
+    Private Sub TestMethod2()
+        Await TestMethod()
+    End Sub
+End Class",
+                GetCA3075LoadBasicResultAt(11, 9, "Run")
+            );
+        }
+
+        [Fact]
+        public void UseXmlDocumentLoadInDelegateShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+class TestClass
+{
+    delegate void Del();
+
+    Del d = delegate () {
+        var xml = """";
+        var doc = new XmlDocument();
+        doc.XmlResolver = null;
+        doc.Load(xml);
+    };
+}",
+                GetCA3075LoadCSharpResultAt(12, 9, "TestClass")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Class TestClass
+    Private Delegate Sub Del()
+
+    Private d As Del = Sub() 
+    Dim xml = """"
+    Dim doc = New XmlDocument()
+    doc.XmlResolver = Nothing
+    doc.Load(xml)
+
+End Sub
+End Class",
+                GetCA3075LoadBasicResultAt(11, 5, "TestClass")
+            );
+        }
+
+        [Fact]
+        public void UseXmlDataDocumentLoadShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    public class TestClass
+    {
+        public void TestMethod(string path)
+        {
+            new XmlDataDocument().Load(path);
+        }
+    }
+}
+",
+                GetCA3075LoadCSharpResultAt(10, 13, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Public Class TestClass
+        Public Sub TestMethod(path As String)
+            Call New XmlDataDocument().Load(path)
```

Comment:
You don't use a `Call` in any of the other tests. (Whenever I see a little difference like that between one test and another similar one, I always have to spend a few seconds asking myself "did that difference actually matter"?)